auth/auth.go

```diff
@@ -708,7 +708,7 @@ func (auth *Authenticator) AuthenticateTrustedJWT(token string, provider *OIDCPr
 	var identity *Identity
 	if provider.AllowUnsignedProviderTokens {
 		// Verify claims - ensures that the token we received from the provider is valid for Sync Gateway
-		identity, err = VerifyClaims(token, provider.ClientID, provider.Issuer)
+		identity, err = VerifyClaims(token, base.StringDefault(provider.ClientID, ""), provider.Issuer)
```

Comment:
Given that it looks like everywhere that uses `provider.ClientID` uses the `StringDefault` util to fall back to empty string - what's the reasoning behind the change to a `*string` ?

Reply:
For local JWT auth, the `client_id` field is required, but can be set to an empty string to disable validation ([spec](https://docs.google.com/document/d/1jxz8fnDfPQ-BXxBGOGEgf2NoTRh00A4v30wPt1NA8hM/edit#heading=h.gyjd8mnd5uke)) - the change is to distinguish the absent case from the empty case.

For OIDC there's no functional change, hence all the defaults, but it'll become meaningful when local JWT authentication is implemented (upcoming PR).